extensions/podman/packages/extension/src/utils/podman-binary.ts

```diff
@@ -0,0 +1,70 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { configuration as configurationAPI, process as processAPI } from '@podman-desktop/api';
+import { injectable } from 'inversify';
+
+import { getPodmanCli } from '/@/utils/podman-cli';
+
+export interface InstalledPodman {
+  version: string;
+}
+
+@injectable()
+export class PodmanBinary {
+  #cli: InstalledPodman | undefined = undefined;
+
+  constructor() {}
+
+  /**
+   * Given a path to a podman binary, return the version of podman
+   * @param path
+   * @protected
+   */
+  protected async getPodmanVersion(path: string): Promise<string> {
+    const { stdout: versionOut } = await processAPI.exec(path, ['--version']);
+    const versionArr = versionOut.split(' ');
+    return versionArr[versionArr.length - 1];
+  }
+
+  public invalidate(): void {
+    this.#cli = undefined;
+  }
+
+  public async getBinaryInfo(): Promise<InstalledPodman | undefined> {
+    if (this.#cli) {
+      return this.#cli;
+    }
+    const path = getPodmanCli();
+    const version = await this.getPodmanVersion(path);
+    this.#cli = {
+      version,
+    };
+    return this.#cli;
+  }
+
+  init(): void {
+    /**
+     * If the user changed the podman binary path, we need to invalidate the cache
+     */
+    configurationAPI.onDidChangeConfiguration(e => {
```

Comment:
to me the issue is more that it's missing to dispose the disposable of the call to onDidChangeConfiguration